### PR TITLE
Allow null TextEditorComponent::domNode during visibility check

### PR DESCRIPTION
**Pull request by as-cii**
...and use `atom.assert` to verify that the constructor was run before the exception.

Ref.: #8762 

/cc: @nathansobo @izuzak

### CODE REVIEW

**Comment by izuzak** [comment]
Thanks for tackling this, @as-cii :zap:

In https://github.com/atom/atom/issues/8762#issuecomment-142209322 you said:

> A package is setting the variable to null. This seems a bit unlikely, though.

Would it make sense to try and confirm or refute that theory as well (if you still think that's a possibility)? I'm not sure, but just wondering if it's possible to add another assert which would catch `@domNode` being set to `null` (e.g. would it be possible to wrap it somehow)?

**Comment by as-cii** [comment]
Thanks for the feedback! :sparkles: 

I still feel like a package is not arbitrarily setting `::domNode` to `null`, but the assertion code is quite straightforward so the tradeoff seems reasonable. :+1: 

Please, have a last :eyes: and :ship: at will. Thanks!

File: src/text-editor-component.coffee [hunk 1]
```diff
@@ -34,6 +34,13 @@ class TextEditorComponent
   stylingChangeAnimationFrameRequested: false
   gutterComponent: null
   mounted: true
+  initialized: false
+
+  Object.defineProperty @prototype, "domNode",
```
Comment (nathansobo) [inline src/text-editor-component.coffee]: Does this perform well? That's my one concern with adding it.

Comment (nathansobo) [inline src/text-editor-component.coffee]: I guess this isn't really a hot code path so whatever.

**Comment by as-cii** [comment]
Thanks!

**Pull request merged**